activerecord/test/cases/finder_test.rb

```diff
@@ -1001,6 +1001,12 @@ def test_hash_condition_local_time_interpolation_with_default_timezone_utc
     end
   end
 
+  def test_condition_arel_attribute
+    relation = Company.where(name: Company.arel_table[:name])
```

Comment:
Why not use `where(Company.arel_table[:name].eq(Company.arel_table[:name]))`?

Reply:
I've since left the company where I was using this although they are still relying on it and I could see myself needing it again. I must admit I can't remember why I didn't do that instead but I am aware that it's possible so there probably was a reason, perhaps that it's not possible in some other contexts? It's a very small but useful change in any case.

Reply:
Adding extra predicate handler makes building all predicate slower.
If `where(Company.arel_table[:name].eq(Company.arel_table[:name]))` is a common use case, people will accept to be slower.
But I don't think that is a common use case, I'm not prefer to be slower by default for making that.

Reply:
Fair enough.